### PR TITLE
Integrate state channel

### PR DESCRIPTION
**Pull request by hackaugusto**

### CODE REVIEW

File: raiden/tests/unit/transfer/mediated_transfer/test_initiator.py [hunk 1]
```diff
@@ -1,585 +0,0 @@
-# -*- coding: utf8 -*-
```
Comment (LefterisJP) [inline raiden/tests/unit/transfer/mediated_transfer/test_initiator.py]: Why do the test_initiator/mediator/target test files go away?

File: raiden/api/python.py [hunk 2]
```diff
@@ -53,501 +47,6 @@
...
-                    'block_number': event.block_number,
-                    '_event_type': type(event.event_object).__name__.encode(),
-                }
-                new_event.update(event.event_object.__dict__)
-                returned_events.append(new_event)
-
-        return returned_events
-
-
 class RaidenAPI2:
```
[the remark this answers is not in the document]
Comment (hackaugusto) [reply raiden/api/python.py]: Yes, I didn't add the rename to reduce the number of changes in this PR. but all the functions / classes that end with a 2 should be renamed.

do you want me to do the renaming in this PR?

File: raiden/tests/unit/transfer/mediated_transfer/test_initiator.py (hunk 1 above)
Comment (hackaugusto) [reply raiden/tests/unit/transfer/mediated_transfer/test_initiator.py]: I added a equivalent version for all of them on a previous PR:

[test_initiatorstate.py)](https://github.com/raiden-network/raiden/blob/e760d0fcc662aaa959aeb0084c0921234f473357/raiden/tests/unit/transfer/mediated_transfer/test_initiatorstate.py)
[test_mediatorstate.py](https://github.com/raiden-network/raiden/blob/e760d0fcc662aaa959aeb0084c0921234f473357/raiden/tests/unit/transfer/mediated_transfer/test_mediatorstate.py)
[test_targetstate.py](https://github.com/raiden-network/raiden/blob/e760d0fcc662aaa959aeb0084c0921234f473357/raiden/tests/unit/transfer/mediated_transfer/test_targetstate.py)

Basically the tests that end on `state` are for the version that uses the ChannelState. Since this PR removes the code that doesnt work with ChannelState their companion tests can go away too.

File: raiden/messages.py [hunk 3]
```diff
@@ -924,7 +914,7 @@ def from_statechange(state_change):
         )
 
     @staticmethod
-    def from_event(event: 'SendMediatedTransfer') -> 'MediatedTransfer':
+    def from_event(event: 'SendMediatedTransfer2') -> 'MediatedTransfer':
```
[the remark this answers is not in the document]
Comment (hackaugusto) [reply raiden/messages.py]: Same thing as above, I didn't rename here to limit the number of changes.

File: raiden/api/python.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (LefterisJP) [reply raiden/api/python.py]: no it's fine if you are going to rename it in the next PR

**Comment by LefterisJP** [comment]
Am satisfied by the answers. Merge when you want. Keep downtime of tests as little as possible.

**Comment by hackaugusto** [comment]
superseded by #1329
